app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryUpdateTest.kt

```diff
@@ -60,4 +65,113 @@ class BloodPressureEntryUpdateTest {
             hasEffects(ShowBpValidationError(ErrorDiastolicEmpty) as BloodPressureEntryEffect)
         ))
   }
+
+  @Test
+  fun `when the date pattern is not valid, then date validation error effect is received`() {
+    val model = defaultModel
+        .dayChanged("32")
+        .yearChanged("9021")
+        .monthChanged("3")
+    spec
+        .given(model)
+        .whenEvents(SaveClicked)
+        .then(
+            assertThatNext(
+                hasNoModel(),
+                hasEffects(
+                    ShowDateValidationError(Invalid.InvalidPattern) as BloodPressureEntryEffect
+                )
+            )
+        )
+  }
+
+  @Test
+  fun `when future date is selected, then show date validation error effect is received`() {
+    val model = defaultModel
+        .dayChanged("12")
+        .yearChanged("2039")
+        .monthChanged("3")
+    spec
+        .given(model)
+        .whenEvents(SaveClicked)
+        .then(
+            assertThatNext(
+                hasNoModel(),
+                hasEffects(
+                    ShowDateValidationError(Invalid.DateIsInFuture) as BloodPressureEntryEffect
+                )
+            )
+        )
+  }
+
+  @Test
+  fun `when the save button is clicked, then show date and BP error effects are received`() {
+    val model = defaultModel
+    spec
+        .given(model)
+        .whenEvents(SaveClicked)
+        .then(
+            assertThatNext(
+                hasNoModel(),
+                hasEffects(
+                    ShowDateValidationError(Invalid.InvalidPattern) as BloodPressureEntryEffect,
+                    ShowBpValidationError(ErrorSystolicEmpty) as BloodPressureEntryEffect
+                )
+            )
+        )
+  }
+
+  @Test
+  fun `when blood pressure is saved, then set blood pressure saved result and dispatch`() {
+    spec
+        .given(defaultModel)
+        .whenEvents(BloodPressureSaved(wasDateChanged = false))
+        .then(
+            assertThatNext(
+                hasModel(defaultModel.bloodPressureStateChanged(BloodPressureSaveState.NOT_SAVING_BLOOD_PRESSURE)),
+                hasEffects(SetBpSavedResultAndFinish as BloodPressureEntryEffect)
+            )
+        )
+  }
+
+  @Test
+  fun `when the save button is clicked, then show no change`() {
```

Comment:
The name of this test does not indicate that the fact that it is testing the condition where a measurement save operation is ongoing. Please update?